browser/extensions/brave_component_loader_browsertest.cc

```diff
@@ -0,0 +1,363 @@
+  }
+
+  void Observe() {
+    // Register a download observer.
+    content::WebContents* contents =
+      browser_->tab_strip_model()->GetActiveWebContents();
+    content::BrowserContext* browser_context = contents->GetBrowserContext();
+    download_manager_ =
+        content::BrowserContext::GetDownloadManager(browser_context);
+    download_manager_->AddObserver(this);
+  }
+
+  void CheckForDownload(const GURL& url) {
+    // Wait for the url to load and check that it loaded.
+    ASSERT_EQ(url, GetLastUrl());
+    // Check that one download has been initiated.
+    DCHECK(download_manager_);
+    std::vector<download::DownloadItem*> downloads;
+    download_manager_->GetAllDownloads(&downloads);
+    ASSERT_EQ(1u, downloads.size());
+    // Cancel download to shut down cleanly.
+    downloads[0]->Cancel(false);
+  }
+
+  const GURL& GetLastUrl() {
+    // Wait until the download has been created.
+    download_run_loop_.Run();
+    return last_url_;
+  }
+
+  // content::DownloadManager::Observer implementation.
+  void OnDownloadCreated(content::DownloadManager* manager,
+                         download::DownloadItem* item) override {
+    last_url_ = item->GetURL();
+    download_run_loop_.Quit();
+  }
+
+ private:
+  Browser* browser_;
+  content::DownloadManager* download_manager_;
+  content::NotificationRegistrar registrar_;
+  base::RunLoop download_run_loop_;
+  GURL last_url_;
+};
+
+class PdfjsExtensionExtensionRegistryObserver
+  : public extensions::ExtensionRegistryObserver {
+public:
+  explicit PdfjsExtensionExtensionRegistryObserver(
+    extensions::ExtensionRegistry* registry)
+    : message_loop_runner_(new content::MessageLoopRunner),
+    added_(false),
+    removed_(false),
+    timeout_(0),
+    elapsed_(0),
+    registry_(registry) {
+    registry_->AddObserver(this);
+  }
+
+  ~PdfjsExtensionExtensionRegistryObserver() override {
+    registry_->RemoveObserver(this);
+  }
+
+  bool added() const { return added_; }
+  bool removed() const { return removed_; }
+
+  bool Wait(int timeout_secs = 0) {
+    if (added_ || removed_)
+      return true;
+    if (timeout_secs) {
+      timeout_ = timeout_secs;
+      elapsed_ = 0;
+      timer_.Start(FROM_HERE, base::TimeDelta::FromSeconds(1), this,
+                   &PdfjsExtensionExtensionRegistryObserver::OnTimer);
+    }
+    message_loop_runner_->Run();
+    return (added_ || removed_);
+  }
+
+  void OnTimer() {
+    if (++elapsed_ >= timeout_ || added_ || removed_)
+      timer_.Stop();
+    if (message_loop_runner_->loop_running())
+      base::ThreadTaskRunnerHandle::Get()->PostTask(
+          FROM_HERE, message_loop_runner_->QuitClosure());
+  }
+
+  void Reset() {
+    added_ = false;
+    removed_ = false;
+  }
+
+  // ExtensionRegistryObserver implementation:
+  void OnExtensionLoaded(content::BrowserContext* browser_context,
+    const extensions::Extension* extension) override {
+    ASSERT_EQ(extension->id(), kTestPdfjsExtensionId);
```

Comment:
I think `ASSERT_STREQ` is more appropriate for string comparisons.

Reply:
Done. Though, I am not sure it's necessary in this case: ASSERT_EQ is templatized, and since both arguments here are std::string it just does std::string::operator== comparison, whereas ASSERT_STREQ explicitly takes const char* parameters.